lib/perl/Genome/Model/ClinSeq.pm

```diff
@@ -249,6 +233,19 @@ sub _resolve_reference {
 sub __errors__ {
     my $self   = shift;
     my @errors = $self->SUPER::__errors__;
+
+    for my $model (qw(wgs_model exome_model)) {
+        if ($self->$model) {
+            unless ($self->$model->isa('Genome::Model::SomaticVariation') || $self->$model->isa('Genome::Model::SomaticValidation')) {
```

Comment:
Our initial thought for the `SomaticInferface` in general was to make it a role, but we ended up going with a class because we wanted to use the built in UR "type checking" on inputs and parameters by updating them to `SomaticInterface`. As far as I know, this would not work if it was a role, but I'm totally ready to be wrong on that! :)